### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by samaid**
Intro section enhancements

### CODE REVIEW

File: dpctl/_sycl_queue_manager.pyx [hunk 1]
```diff
@@ -213,7 +213,7 @@ cpdef get_current_backend():
 @contextmanager
 def device_context(arg):
     """
-    Yields a SYCL queue corresponding to the input filter string.
+    Yields a SYCL queue corresponding to the input device object, the queue object, or the filter string.
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_queue_manager.pyx:216]:
```suggestion
    Yields a SYCL queue corresponding to the input device object, queue object, or filter string.
```

File: dpctl/_sycl_queue_manager.pyx [hunk 2]
```diff
@@ -223,25 +223,22 @@ def device_context(arg):
     usable queue on exiting the context manager.
 
     Args:
-
-        queue_str (str) : A string corresponding to the DPC++ filter selector.
+        arg : A :class:`.SyclDevice` object, or a :class:`.SyclQueue` object, or a filter selector string
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_queue_manager.pyx:226]:
```suggestion
        arg : A :class:`dpctl.SyclQueue` object, or a :class:`dpctl.SyclDevice` object, or a filter selector string
```

File: dpctl/_sycl_queue_manager.pyx [hunk 3]
```diff
@@ -223,25 +223,22 @@ def device_context(arg):
...
-        filter string.
+        :class:`.SyclQueue`: A SYCL queue corresponding to the specified input device, queue, or filter string.
 
     Raises:
         SyclQueueCreationError: If the SYCL queue creation failed.
 
     :Example:
-        To create a scope within which the Level Zero GPU number 0 is active,
-        a programmer needs to do the following.
+        The following example activates Level Zero GPU number 0 by respective filter selector string:
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_queue_manager.pyx:235]:
```suggestion
        The following example sets current queue targeting specific device indicated with filter selector string in the scope of `with` block:
```

File: dpctl/_sycl_queue_manager.pyx (hunk 1 above)
Comment (oleksandr-pavlyk) [reply dpctl/_sycl_queue_manager.pyx:216]: Reordered enumeration of argument types: explicit queue instance, explicit device instance, device filter selector string.

File: dpctl/_sycl_queue_manager.pyx (hunk 3 above)
Comment (oleksandr-pavlyk) [reply dpctl/_sycl_queue_manager.pyx:235]: Note that the whole concept of `device_context` and queue manager is de-facto deprecated and will become obsolete once compute follows data is embraced.

Comment (samaid) [reply dpctl/_sycl_queue_manager.pyx:235]: @oleksandr-pavlyk, yes, the compute follows data will make device_context() almost obsolete. But what about scenario when I created NumPy array with NumPy on the host but want the kernel to be compiled for GPU with Numba, data inherently converted from NumPy array to underlying USM array?

import dpctl
import numpy as np
from numba import njit

@njit
def do_compute(a):
    return np.log(a+1.0)

a = np.arange(10)
with dpctl.device_context(dpctl.select_gpu_device())
    b = do_compute(a)
print(b)

I find it very convenient. I remember multiple discussions about the compute follow data but do not remember we agreed on deprecation of the device context mechanism.

**Comment by oleksandr-pavlyk** [comment]
Split this RP into 5 separate PRs: #659, #660, #661, #662, #663 separate by the filename changed.

This PR can be now closed.

**Pull request closed without merging**